gnosis/cowswap
# Handle cancel state for ethereum txs

# Summary

Closes #1382 #1435

This adds a `Cancelling` and `Cancelled` states for ethereum transactions in the Recent Activity list. Before cancelled txs were removed from the list when they were confirmed.

The changes are built on top of @anxolin's PR https://github.com/gnosis/cowswap/pull/1429 as I'm making use of the new `EnhancedTransactionDetails` type and also to reduce conflicts due to changes in the `enhancedTransactions` state files.

![image](https://user-images.githubusercontent.com/3328006/136390642-914fd511-028b-485d-bdb3-085c00673cbc.png)

  # To Test

1. Run an onchain transaction (wrap eth or approve) with very low gas fees
2. Cancel it through MM 
3. The transaction log in the Recent Activity list should change to `Cancelling` and then `Cancelled` when confirmed
